contracts/network/BancorNetwork.sol

```diff
@@ -579,6 +607,41 @@ contract BancorNetwork is IBancorNetwork, Upgradeable, ReentrancyGuardUpgradeabl
         return _withdrawBaseToken(contextId, provider, completedRequest);
     }
 
+    /**
+     * @inheritdoc IBancorNetwork
+     */
+    function withdrawSurplusTokens(Token pool) external whenNotPaused nonReentrant returns (uint256) {
+        // verify pool collection exists and retrieve it
+        IPoolCollection poolCollection = _poolCollection(pool);
```

Comment:
might want to add:

`if (!_networkSettings.isTokenWhitelisted(token)) {
            revert NotWhitelisted();
 }`

Reply:
Added

Reply:
Where is this check?